### ISSUE TITLE
Possible Issue routing does not follow highway (-link)

### ISSUE THREAD

**Issue by Tobey16** [organicmaps/organicmaps#788]
There may be an error in the router. Im working on a older branch, and there it could not find a route from
https://www.openstreetmap.org/#map=18/49.35220/11.20052&layers=N
to
https://www.openstreetmap.org/#map=18/49.34577/11.20365&layers=N

I already have made a changeset in OSM, that i suspected would be the cause.

Please let me know if you can reproduce this error on the current version.

**Comment by biodranik** [comment]
It reproduces in our current dataset from 29.05.2021. When did you edit it?

**Comment by Tobey16** [comment]
I edited it on the 8. of june. But the error still exists. Too bad that there is not much documentation. I would have looked deeper into this.

**Comment by biodranik** [comment]
Then obviously it won't work until the next maps data update.

**Comment by Tobey16** [comment]
I tried it with my branch (repo). It did not change the outcome. it must be something else. I wish i could investigate this error, but without any documentation, it would take a lot of time to just find out how the router works.

**Comment by vng** [comment]
Well, seems like it's a bug in routing. The data is ok and it always was ok. I suspect that problem with algorithm when more than 1 links come out from any highway ..

Thank you for the report.

**Comment by Tobey16** [comment]
It was no bug in version 9.4.7. It  must have been introduced.

**Comment by biodranik** [comment]
Are you checking it in Organic Maps or in MapsMe?

**Comment by Tobey16** [comment]
It is MapsMe. Dont they both share the same branch?

**Comment by biodranik** [comment]
No. Here we're working on Organic Maps.

**Comment by Tobey16** [comment]
> It reproduces in our current dataset from 29.05.2021. When did you edit it?

You said it has the same bug (in other words). I call this "very similiar". It is much more useful to discuss the bug, than what app it exactly is. I would offer my time to investigate this bug, but without some help i cant do much.

**Comment by biodranik** [comment]
No, that was another issue. We have updated OSM data snapshot in the recent release of Organic Maps, did you check this bug there?

**Comment by vng** [comment]
I investigated this issue. This way https://www.openstreetmap.org/way/221960008 and incoming junction and all the A9 (North-South) is treated as dead-end road for the algo. It goes down until here https://www.openstreetmap.org/#map=18/49.30849/11.20728 where new mwm Bavaria_Upper Palatinate begins. After that again Bavaria_Middle Franconia. Obvious bug in algorithm but this is a bit tricky situation and I don't see how to make some safe fix now ..

As a workaround - build route that **not** starting and ending at this speed highway.

Check for IndexRouter::FindBestEdges, EraseIfDeadEnd and IsDeadEndCached calls there ..

**Comment by Jean-BaptisteC** [comment]
@Tobey16 can you check on latest version?
